src/ngx_http_lua_kong_ssl.c

```diff
@@ -179,6 +179,42 @@ ngx_http_lua_kong_ffi_request_client_certificate(ngx_http_request_t *r)
 }
 
 
+/*
+ * Set the CA DN list to the underlying SSL structure, which will be sent in the
+ * Certificate Request Message of downstram TLS handshake.
+ *
+ * The downstream client can use this DN information to filter certificates,
+ * and chooses an appropriate certificate issued by a CA in the list.
+ *
+ * on success, NULL is returned, otherwise a static string indicating the
+ * failure reason is returned.
+ */
+
+const char *
+ngx_http_lua_kong_ffi_set_client_ca_list(ngx_http_request_t *r,
+                                         STACK_OF(X509_NAME) *name_list)
+{
+#if (NGX_SSL)
+    ngx_connection_t    *c = r->connection;
+    ngx_ssl_conn_t      *sc;
+
+    if (c->ssl == NULL) {
+        return "server does not have TLS enabled";
+    }
+
+    sc = c->ssl->connection;
+
+
+    SSL_set_client_CA_list(sc, name_list);
```

Comment:
I also noticed this and deep copied the `name_list` in the outer layer. I think you're right, it's better to do this within the interface. BTW, `X509_chain_up_ref` works on `STACK_OF(X509) *`, and `X509_NAME` has no reference counter, so we can only deep copy it.

Reply:
ah yes, i misread the signature, it's a `STACK_OF(X509_NAME)`. however, there isn't a convient function in openssl that creates a `STACK_OF(X509_NAME)`, maybe possible with some development with `resty.openssl.stack`, but that makes this API not immediately available.

so i would suggest a bit change to the API, instead of accepting a `STACK_OF(X509_NAME)`, let's pass a `STACK_OF(X509)` (which is then a `resty.openssl.x509.chain`.

in `ngx_http_lua_kong_ffi_set_client_ca_list`, we will 1) create a stack_of(x509_name), 2) iterate over the x509 stack, and generate X509_NAME from each of the cert, 3) add to the stack, 4) call SSL_set_client_CA_list. don't need to free the stack or the x509_name, it will be freed together with the `SSL` instance.

Reply:
Good suggestion! Taking `STACK_OF(X509) *` as a parameter does make the lua code simpler and thus alleviates the burden of users.